internal/pkg/api/handleCheckin.go

```diff
@@ -500,6 +500,10 @@ func (ct *CheckinT) processUpgradeDetails(ctx context.Context, agent *model.Agen
 	doc := bulk.UpdateFields{
 		dl.FieldUpgradeDetails: details,
 	}
+	if agent.UpgradeAttempts != nil && details.State == UpgradeDetailsStateUPGWATCHING {
+		doc[dl.FieldUpgradeAttempts] = nil
```

Comment:
Just to be clear, this will set `upgrade_attempts` to `nil`, it does not remove the attribute from the agent doc. This is the intended behaviour, right?

Reply:
we want to set it to `null`